archivistcrawler/networkConfig.nim

```diff
@@ -0,0 +1,143 @@
+import std/envvars
+import std/httpclient
+import std/sequtils
+
+import pkg/serde/json
+import pkg/questionable
+import pkg/questionable/results
+
+# Endpoint types
+type
+  NetworkConfig* = object
+    latest* {.serialize.}: string
+    archivist* {.serialize.}: seq[ArchivistVersionEntry]
+    sprs* {.serialize.}: seq[ArchivistSprEntry]
+    rpcs* {.serialize.}: seq[string]
+    marketplace* {.serialize.}: seq[ArchivistMarketplaceEntry]
+    team* {.serialize.}: ArchivistNetworkTeamObject
+
+  ArchivistVersionEntry* = object
+    version* {.serialize.}: string
+    revision* {.serialize.}: string
+    contracts* {.serialize.}: string
+
+  ArchivistSprEntry* = object
+    supportedVersions* {.serialize.}: seq[string]
+    records* {.serialize.}: seq[string]
+
+  ArchivistMarketplaceEntry* = object
+    supportedVersions* {.serialize.}: seq[string]
+    contractAddress* {.serialize.}: string
+    abi* {.serialize.}: string
+
+  ArchivistNetworkTeamObject* = object
+    nodes* {.serialize.}: seq[ArchivistNetworkTeamNodesEntry]
+    utils* {.serialize.}: ArchivistNetworkTeamUtilsObject
+
+  ArchivistNetworkTeamNodesEntry* = object
+    category* {.serialize.}: string
+    versions* {.serialize.}: seq[ArchivistNetworkTeamNodesVersionsEntry]
+
+  ArchivistNetworkTeamNodesVersionsEntry* = object
+    version* {.serialize.}: string
+    instances* {.serialize.}: seq[ArchivistNetworkTeamNodesVersionsInstancesEntry]
+
+  ArchivistNetworkTeamNodesVersionsInstancesEntry* = object
+    name* {.serialize.}: string
+    podName* {.serialize.}: string
+    ethAddress* {.serialize.}: string
+
+  ArchivistNetworkTeamUtilsObject* = object
+    crawlerRpc* {.serialize.}: string
+    botRpc* {.serialize.}: string
+    elasticSearch* {.serialize.}: string
+
+# Application types
+type
+  ArchivistNetwork* = object
+    version* {.serialize.}: ArchivistVersionEntry
+    spr* {.serialize.}: ArchivistSprEntry
+    rpcs* {.serialize.}: seq[string]
+    marketplace* {.serialize.}: ArchivistMarketplaceEntry
+    team* {.serialize.}: TeamObject
+
+  TeamObject* = object
```

Comment:
These team nodes are not used anywhere in the crawler, so I wouldn't deserialize them here either. It just gives you extra maintenance of unused code if the json format changes.